### PR TITLE
add GetTotalSupply API

### PR DESCRIPTION
**Pull request by KaustubhShamshery**
## Description
Add API to get total coin supply

## Backward Compatibility

- [x] This is not a breaking change
- [ ] This is a breaking change

### Implementation

- [ ] **ready for review**

### Integration Test (Core Team)

- [x] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

File: src/common/Constants.h [hunk 1]
```diff
@@ -69,6 +69,8 @@ const unsigned int INIT_DS_EPOCH_NUM = 2;
 
 const unsigned int MAINNET_CHAIN_ID = 1;
 
+const uint64_t MAX_COIN_SUPPLY = 12000000000;  // 12 Billion
```
Comment (bb111189) [inline src/common/Constants.h]: should be 21 billions

Comment (bb111189) [reply src/common/Constants.h]: TOTAL_COINBASE_REWARD + TOTAL_GENESIS_TOKEN

File: src/libServer/LookupServer.cpp [hunk 2]
```diff
@@ -1075,6 +1079,17 @@ double LookupServer::GetTxBlockRate() {
   return ans.convert_to<double>();
 }
 
+double LookupServer::GetTotalCoinSupply() {
+  boost::multiprecision::cpp_dec_float_50 ans(
+      to_string(MAX_COIN_SUPPLY));  // convert to QA
+  const Account* account = AccountStore::GetInstance().GetAccount(NullAddress);
+  boost::multiprecision::cpp_dec_float_50 rewards(account->GetBalance().str());
+  rewards /= 1000000000000;
```
Comment (bb111189) [inline src/libServer/LookupServer.cpp]: let's do safemath here

File: src/libServer/LookupServer.cpp [hunk 3]
```diff
@@ -1075,6 +1079,17 @@ double LookupServer::GetTxBlockRate() {
...
+  ans -= rewards;
```
Comment (bb111189) [inline src/libServer/LookupServer.cpp]: let's do safemath here

File: src/libServer/LookupServer.cpp [hunk 4]
```diff
@@ -1075,6 +1079,17 @@ double LookupServer::GetTxBlockRate() {
   return ans.convert_to<double>();
 }
 
+double LookupServer::GetTotalCoinSupply() {
+  boost::multiprecision::cpp_dec_float_50 ans(
+      to_string(MAX_COIN_SUPPLY));  // convert to QA
```
Comment (ansnunez) [inline src/libServer/LookupServer.cpp]: I tried without the cast to string, still seems to work

**Pull request merged**